### DISCUSSION TITLE
Add ability to store migration version in a different database than the one that is migrated, or in a file

### DISCUSSION THREAD

**Discussion by ctacat** [golang-migrate/migrate#1147] [Ideas]
Hello,

I'm happily using golang-migrate currently (as a library) to carry my migration needs. For a particular case, I need to store the migration version outside of the database that has to be migrated, or in a file.

I can't find any way with the current release to do that (neither did I find any other tool able to). So I'm going to fork golang-migrate to implement this. My idea is to introduce a VersionDrv struct (similar in spirit to DatabaseDrv or SourceDrv) to define how to handle version storage.

If you think that it's something that should be integrated in official release, let me know; I will submit a PR to do so. I will accept also any suggestions about the way to implement this accordingly with current architecture guidelines.

Best regards.

**Comment by dsyers** [comment]
Not sure if this would be of use: https://github.com/golang-migrate/migrate/pull/1294
